lib/Launcher.js

```diff
@@ -48,62 +48,50 @@ const DEFAULT_ARGS = [
   '--metrics-recording-only',
   '--no-first-run',
   '--safebrowsing-disable-auto-update',
-];
-
-const AUTOMATION_ARGS = [
   '--enable-automation',
   '--password-store=basic',
   '--use-mock-keychain',
 ];
 
 class Launcher {
   /**
-   * @param {!(LaunchOptions & BrowserOptions)=} options
+   * @param {!(LaunchOptions & ChromeArgOptions & BrowserOptions)=} options
    * @return {!Promise<!Browser>}
    */
-  static async launch(options) {
-    options = Object.assign({}, options || {});
-    let temporaryUserDataDir = null;
-    const chromeArguments = [];
-    if (!options.ignoreDefaultArgs)
-      chromeArguments.push(...DEFAULT_ARGS);
-    if (!options.ignoreDefaultArgs)
-      chromeArguments.push(...AUTOMATION_ARGS);
-
-    if (!options.ignoreDefaultArgs || !chromeArguments.some(argument => argument.startsWith('--remote-debugging-')))
-      chromeArguments.push(options.pipe ? '--remote-debugging-pipe' : '--remote-debugging-port=0');
+  static async launch(options = {}) {
+    const {
+      ignoreDefaultArgs = false,
+      args = [],
+      dumpio = false,
+      executablePath = null,
+      pipe = false,
+      env = process.env,
+      handleSIGINT = true,
+      handleSIGTERM = true,
+      handleSIGHUP = true,
+      ignoreHTTPSErrors = false,
+      defaultViewport = {width: 800, height: 600},
+      slowMo = 0,
+      timeout = 30000
+    } = options;
 
-    if (!options.args || !options.args.some(arg => arg.startsWith('--user-data-dir'))) {
-      if (!options.userDataDir)
-        temporaryUserDataDir = await mkdtempAsync(CHROME_PROFILE_PATH);
+    const chromeArguments = !ignoreDefaultArgs ? this.defaultArgs(options) : args;
+    let temporaryUserDataDir = null;
 
-      chromeArguments.push(`--user-data-dir=${options.userDataDir || temporaryUserDataDir}`);
+    if (!chromeArguments.some(argument => argument.startsWith('--remote-debugging-')))
+      chromeArguments.push(pipe ? '--remote-debugging-pipe' : '--remote-debugging-port=0');
+    if (!ignoreDefaultArgs && !chromeArguments.some(arg => arg.startsWith('--user-data-dir'))) {
```

Comment:
I don't think we want `ignoreDefaultArgs` to disable temp userDataDir management. We didn't do this before.

Reply:
Fair.